src/orchestrator/types/detection_batcher/chat_completion.rs

```diff
@@ -14,48 +14,590 @@
  limitations under the License.
 
 */
-#![allow(dead_code)]
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, btree_map};
 
 use super::{Chunk, DetectionBatcher, Detections, DetectorId, InputId};
 
 /// A batcher for chat completions.
+///
+/// A batch corresponds to a chunk. Batches are returned in-order
+/// as detections from all detectors are received for the chunk.
+/// For chat completions, each chunk will also correspond to a choice,
+/// where choices are independent of each other. Each streamed event
+/// for chat completions has a 'choices' field, but only contains
+/// one choice per event, e.g.
+/// data: {"id":"chat-", ..., "choices":[{"index":0, ...}]}
+/// data: {"id":"chat-", ..., "choices":[{"index":1, ...}]}
+/// data: {"id":"chat-", ..., "choices":[{"index":0, ...}]}
+///
+/// The `choice_index` can then be tracked with each chunk,
+/// but do not have to be batched or aggregated for stream events.
+///
+/// This batcher requires that all detectors use the same chunker,
+/// like the max_processed_index batcher. Detections from detectors
+/// that require whole input and/or output of a model (i.e. that
+/// do not require chunking with chunkers) will be handled separately.
```

Comment:
I found this a little confusing to read, so would suggest we refer to these as "choice-chunk" (pairs) and reword the docstring to:

```suggestion
/// A batcher for chat completions.
///
/// A batch corresponds to a choice-chunk. Batches are
/// returned in-order as detections from all detectors
/// are received for the choice-chunk.
///
/// Chat completion messages have a `choices` field containing
/// a single choice, e.g.
/// ```text
///     data: {"id":"chat-", ..., "choices":[{"index":0, ...}]}
///     data: {"id":"chat-", ..., "choices":[{"index":1, ...}]}
/// ```
/// And we track chunks for each choice independently.
///
/// This batcher requires that all detectors use the same chunker.
```

The note about non-streaming/whole_doc_chunker detections isn't relevant to the batcher, so I removed that from here.